### ISSUE TITLE
Unable To list subscriptions after login

### ISSUE THREAD

**Issue by reallydontask** [Azure/azure-cli#8734]
**Describe the bug**

I get the following error after logging in.

```
PS > az login
Note, we have launched a browser for you to login. For old experience with device code, use "az login --use-device-code"
You have logged in. Now let us find all the subscriptions to which you have access...

Error occurred in request., ConnectionError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x04D537D0>: Failed to establish a new connection: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond',))
Traceback (most recent call last):
  File "C:\Users\VssAdministrator\AppData\Local\Temp\pip-install-bsrj0zf4\urllib3\urllib3\connection.py", line 159, in _new_conn
  File "C:\Users\VssAdministrator\AppData\Local\Temp\pip-install-bsrj0zf4\urllib3\urllib3\util\connection.py", line 80, in create_connection
  File "C:\Users\VssAdministrator\AppData\Local\Temp\pip-install-bsrj0zf4\urllib3\urllib3\util\connection.py", line 70, in create_connection
TimeoutError: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond
```
If I set https_proxy variable :
```

PS > az login
Note, we have launched a browser for you to login. For old experience with device code, use "az login --use-device-code"
You have logged in. Now let us find all the subscriptions to which you have access...
Error occurred in request., ProxyError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by ProxyError('Cannot connect to proxy.', OSError('Tunnel connection failed: 407 Proxy Authentication Required',)))
Traceback (most recent call last):
  File "C:\Users\VssAdministrator\AppData\Local\Temp\pip-install-bsrj0zf4\urllib3\urllib3\connectionpool.py", line 594, in urlopen
  File "C:\Users\VssAdministrator\AppData\Local\Temp\pip-install-bsrj0zf4\urllib3\urllib3\connectionpool.py", line 805, in _prepare_proxy
  File "C:\Users\VssAdministrator\AppData\Local\Temp\pip-install-bsrj0zf4\urllib3\urllib3\connection.py", line 308, in connect
  File "C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\http\client.py", line 919, in _tunnel
    message.strip()))
OSError: Tunnel connection failed: 407 Proxy Authentication Required

```
If I set https_proxy variable with the **wrong** proxy (i.e. non existent ip address) then the error is this:
```

PS > az login
Note, we have launched a browser for you to login. For old experience with device code, use "az login --use-device-code"

Please ensure you have network connection. Error detail: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /common/oauth2/token (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x052ABEF0>: Failed to establish a new connection: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond',)))

```

I have closed and reopened powershell after changes to env variables

The behaviour is the same on the command prompt

**To Reproduce**

1. Az Login

1. Select appropriate account on browser window

1. Wait for error

**Expected behavior**

it should list the subscriptions i have access to

**Environment summary**

Windows 10 - 1803 - 17134.590
Used Msi file to install Az CLI
version - 2.0.59 - This is a guess based on the msi, az --version commands doesn't seem to provide any output (5 minutes and counting)
I have tried from both Command Prompt and PowerShell

I am behind a corporate proxy

**Additional context**

**Comment by marstr** [comment]
What happens if you run the following?

``` bash
az login --use-device-code
```

I suspect that it's only the loop-back to the server we spin-up on localhost to complete the OAuth dance that is getting blocked.

**Comment by reallydontask** [comment]
I get a similar error:
`
Error occurred in request., ConnectionError: HTTPSConnectionPool(host='management.azure.com', port=443): Max retries exceeded with url: /tenants?api-version=2016-06-01 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x04C320D0>: Failed to establish a new connection: [WinError 10060] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond',))`

**Comment by marstr** [comment]
Interesting, it's not the error that I thought then, and it is distinct from #7227.

**Comment by marstr** [comment]
The thing that's puzzling to me is that it seems like you're able to authenticate but then get blocked by your proxy for future requests. (Which probably means the Authentication label should come off of this, but I don't have a better fit for it.) @yugangw-msft, have you seen anything like this in the past?

**Comment by marstr** [comment]
Looks like there is some existing guidance available here: https://github.com/Azure/azure-cli/blob/dev/doc/use_cli_effectively.md#working-behind-a-proxy

Also, it looks like this is a duplicate of https://github.com/Azure/azure-cli/issues/8632, so I'm going to close it out.

**Comment by reallydontask** [comment]
For future reference. This is what i had to do

1. Open Powershell with elevated permissions

1. Run Following commands:

[environment]::SetEnvironmentVariable("HTTPS_PROXY","http://<domain>5%C<username>:<password>:<proxyip>:<proxyport>","Machine")

[environment]::SetEnvironmentVariable("AZURE_CLI_DISABLE_CONNECTION_VERIFICATION",1,"Machine")

1. Close PowerShell

1. Open PowerShell 

1. az login

This will give you warnings but it will work

**Comment by rinaldochristy** [comment]
@reallydontask so you had a proxy and hence it was not working, I do not have any proxies but I have the exact same issue as you had! What can I do?

**Comment by BrainCrumbz** [comment]
@DreadKong by any chance were you able to overcome this issue? We're in a work-from-home setup and one of our colleagues home laptop has this issue.

Machine seems _not_ behind a proxy: public IP address on [www.whatismyip](https://www.whatismyip.com/) matches the router IP as seen from router configuration, as well as [whatismyproxy](http://www.whatismyproxy.com/) says "No proxies were detected.". Also, from web search, network provider does not seem to setup any proxy on its own.
